--- Cloak/internal/server/usermanager/api_router.go.orig	2023-08-18 18:09:28.619842711 +0300
+++ Cloak/internal/server/usermanager/api_router.go	2023-08-18 18:27:02.778903975 +0300
@@ -5,6 +5,7 @@ import (
 	"encoding/base64"
 	"encoding/json"
 	"net/http"
+        "math"
 
 	gmux "github.com/gorilla/mux"
 )
@@ -102,6 +103,30 @@ func (ar *APIRouter) writeUserInfoHlr(w
 		http.Error(w, "UID mismatch", http.StatusBadRequest)
 	}
 
+        if uinfo.SessionsCap == nil || *uinfo.SessionsCap == 0 {
+                uinfo.SessionsCap = 1
+        }
+
+        if uinfo.UpRate == nil || *uinfo.UpRate == 0 {
+                uinfo.UpRate = math.MaxInt64
+        }
+
+        if uinfo.DownRate == nil || *uinfo.DownRate == 0 {
+                uinfo.DownRate = math.MaxInt64
+        }
+
+        if uinfo.UpCredit == nil || *uinfo.UpCredit == 0 {
+                uinfo.UpCredit = math.MaxInt64
+        }
+
+        if uinfo.DownCredit == nil || *uinfo.DownCredit == 0 {
+                uinfo.DownCredit = math.MaxInt64
+        }
+
+        if uinfo.ExpiryTime == nil || *uinfo.ExpiryTime == 0 {
+                uinfo.ExpiryTime = math.MaxInt64
+        }
+
 	err = ar.manager.WriteUserInfo(uinfo)
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
